### PR TITLE
(370) Move the new note form onto its own page

### PR DESCRIPTION
**Pull request by jkempster34**
Currently, the new note form is underneath the project summary and subnavigation. This means that the error summary sits halfway down the page, whereas the GOV.UK guidance says it should sit at the top.

After a chat with our interaction designer, we have decided to move the new note form onto its own page. This solves the error message issue, and has an additional benefit that editing messages will be easier if we choose to implement that in the future.

This includes a few style updates:
- Add a cancel link to the new note form
- Rename the textarea label from "Notes" to "Enter note"
- Change the notes form submission button from "Add note" to "Save note"
- Remove the top separator of the list of notes

## Screenshots
image
image
image

## Checklist

- [x] Attach this pull request to the appropriate card in Trello.
- [x] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

**Comment by jkempster34** [comment]
[Improve the error summary for the new note form](https://trello.com/c/mTiHWyTS/370-improve-the-error-summary-for-the-new-note-form)

File: app/views/notes/_notes.html.erb [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
     <span class="govuk-caption-m"><%= note.created_at.to_date.to_formatted_s(:govuk) %></span>
-
     <h3 class="govuk-heading-m"><%= note.user.email %></h3>
 
     <p class="govuk-body"><%= note.body %></p>
   <% end %>
 <% else %>
-  <hr class="govuk-section-break govuk-section-break--xl govuk-section-break--visible">
-
   <h3 class="govuk-heading-m"><%= t("note.index.no_notes_yet") %></h3>
```
Comment (jacksonj04) [inline app/views/notes/_notes.html.erb:13]: This doesn't feel quite right as a header now that there isn't a form on the page. Maybe an [inset text](https://design-system.service.gov.uk/components/inset-text/) above the "add form" button would be cleaner?

**Review by jacksonj04 — CHANGES_REQUESTED** [review]
Couple of tiny picky bits; substance of the change looks sound to me :shipit:

**Review by jacksonj04 — APPROVED** [review]
:shipit:

**Pull request merged**